crates/sui/src/client_ptb/lexer.rs

```diff
@@ -279,12 +279,24 @@ impl<'l, I: Iterator<Item = &'l str>> Iterator for Lexer<'l, I> {
             }
 
             sp!(_, "-") => 'command: {
-                let Some(prefix) = self.eat_prefix("--") else {
+                self.bump();
+                let Some(next) = self.peek() else {
                     break 'command self.unexpected(c);
                 };
 
+                match next {
+                    sp!(_, "-") => {
+                        self.bump();
+                    }
+                    sp!(_, flag) if is_flag(flag.chars().next().unwrap()) => {
```

Comment:
Is it guaranteed that this `next` will produce a `Some`?

Reply:
Yes, `peek` only returns `Some(...)` if there is a character to peek, so `next` will always contain exactly one character.